File: 5_kyu/RGB_To_Hex_Conversion.lua
local kata = {}

function kata.rgb(r, g, b)
    local function cut(x)
        if x <= 0 then
            return 0
        elseif x >= 255 then
            return 255
        else
            return x
        end
    end
    return string.format("%02X%02X%02X", cut(r), cut(g), cut(b) )
end

return kata

File: 5_kyu/RGB_To_Hex_Conversion.rs
#[inline]
fn clamp_to_u8(x: i32) -> u8 {
    x.clamp(0,255) as u8
}

fn rgb(r: i32, g: i32, b: i32) -> String {
    format!("{:02X}{:02X}{:02X}",clamp_to_u8(r),clamp_to_u8(g),clamp_to_u8(b))
}
